full-service/src/service/receipt.rs

```diff
@@ -591,7 +592,7 @@ mod tests {
         // Land the Txo in the ledger - only sync for the sender
         TransactionLog::log_submitted(
             &tx_proposal,
-            14,
```

Comment:
this tx actually lands in block_index 13 and the test conditions never create a block_index 14. Improvements to the ledger->wallet sync code cause this test to fail now due to the off-by-one error in the logging call.

The update to the test makes a real-time determination of which block_index to use.

There are two more identical updates below for the same reason.